### PR TITLE
Diff modal

### PR DESCRIPTION
**Pull request by sdess09**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Other (please describe):

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] If it's a backend change, tests for the changes have been added and `go test ./...` runs successfully from the root folder.
- [ ] If it's a frontend change, Prettier has been run
- [ ] Docs have been reviewed and added / updated if needed

### CODE REVIEW

File: dashboard/src/main/home/app-dashboard/expanded-app/ChangeLogModal.tsx [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+import api from "shared/api";
+import { Context } from "shared/Context";
+
+type Props = {
+  modalVisible: boolean;
+  setModalVisible: (x: boolean) => void;
+  revision: number;
+  currentChart: ChartType;
+
+  //envChild: any;
```
Comment (rgbmdq) [inline dashboard/src/main/home/app-dashboard/expanded-app/ChangeLogModal.tsx:34]: remove commented code

File: dashboard/src/main/home/app-dashboard/expanded-app/ChangeLogModal.tsx [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+};
+
+const ChangeLogModal: React.FC<Props> = ({
+  revision,
+  currentChart,
+  modalVisible,
+  //envChild,
+  setModalVisible,
+}) => {
+  const [scrollToBottomEnabled, setScrollToBottomEnabled] = useState(true);
```
Comment (rgbmdq) [inline dashboard/src/main/home/app-dashboard/expanded-app/ChangeLogModal.tsx:44]: sa,e here

File: dashboard/src/main/home/app-dashboard/expanded-app/ChangeLogModal.tsx [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+  const [currentView, setCurrentView] = useState("overview");
```
Comment (rgbmdq) [inline dashboard/src/main/home/app-dashboard/expanded-app/ChangeLogModal.tsx:45]: perhaps "overview" should be a constant

File: dashboard/src/main/home/app-dashboard/expanded-app/ChangeLogModal.tsx [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+      {
+        name: chart.name,
+        namespace: chart.namespace,
+        cluster_id: currentCluster.id,
+        revision: revision,
+        id: currentProject.id,
+      }
+    );
+    const updatedChart = res.data;
+    //console.log(updatedChart);
```
Comment (rgbmdq) [inline dashboard/src/main/home/app-dashboard/expanded-app/ChangeLogModal.tsx:77]: remove

File: dashboard/src/main/home/app-dashboard/expanded-app/ChangeLogModal.tsx [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+    };
+
+    fetchData();
+  }, [currentChart.config]);
+  const parseYamlAndDisplayDifferences = (oldYaml: any, newYaml: any) => {
+    const diff = Diff.diff(oldYaml, newYaml);
+    const changes: JSX.Element[] = [];
+
+    // Define the regex pattern to match service creation
+    const servicePattern = /^[a-zA-Z0-9\-]*-[a-zA-Z0-9]*[^\.]$/;
```
Comment (rgbmdq) [inline dashboard/src/main/home/app-dashboard/expanded-app/ChangeLogModal.tsx:103]: if this regex it's going to be used several times, it wouldn't be better have it on a helper field?

**Comment by jusrhee** [comment]
Few minor notes:
- Let's call it "View changes" instead of "View change log"
- Know we need split view but could we make the modal a bit less wide since the bullets are the default (maybe the accountsettings modal +100px?)
- Spacing on "Show raw diff" is a bit weird so let's just put it inline at the bottom:
image
- As a general reference, we only capitalize the first letter even for headings (a la GitHub), so checkbox text should be "Show raw diff"
- Let's include the revision diff we're showing in the modal title (i.e., "Changes for version 3" instead of "Change Log" and "Revert to version 9")

**Review by jusrhee — APPROVED** [review]

**Pull request merged**